apps/studio/components/interfaces/Storage/EditBucketModal.tsx

```diff
@@ -16,256 +42,332 @@ import { useBucketUpdateMutation } from 'data/storage/bucket-update-mutation'
 import { IS_PLATFORM } from 'lib/constants'
 import { Admonition } from 'ui-patterns'
 import { Bucket } from 'data/storage/buckets-query'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+import { isNonNullable } from 'lib/isNonNullable'
 
 export interface EditBucketModalProps {
-  visible: boolean
-  bucket?: Bucket
+  bucket: Bucket
   onClose: () => void
 }
 
-const EditBucketModal = ({ visible, bucket, onClose }: EditBucketModalProps) => {
+const BucketSchema = z.object({
+  name: z.string(),
+  public: z.boolean().default(false),
+  has_file_size_limit: z.boolean().default(false),
+  formatted_size_limit: z.coerce
+    .number()
+    .min(0, 'File size upload limit has to be at least 0')
+    .default(0),
+  allowed_mime_types: z.string().trim().default(''),
+})
+
+const formId = 'edit-storage-bucket-form'
+
+export const EditBucketModal = ({ bucket, onClose }: EditBucketModalProps) => {
   const { ref } = useParams()
 
-  const { mutate: updateBucket, isLoading: isUpdating } = useBucketUpdateMutation({
-    onSuccess: () => {
-      toast.success(`Successfully updated bucket "${bucket?.name}"`)
-      onClose()
-    },
-  })
-  const { data } = useProjectStorageConfigQuery(
-    { projectRef: ref },
-    { enabled: IS_PLATFORM && visible }
-  )
+  const { mutate: updateBucket, isLoading: isUpdating } = useBucketUpdateMutation()
+  const { data } = useProjectStorageConfigQuery({ projectRef: ref }, { enabled: IS_PLATFORM })
   const { value, unit } = convertFromBytes(data?.fileSizeLimit ?? 0)
   const formattedGlobalUploadLimit = `${value} ${unit}`
 
-  const [selectedUnit, setSelectedUnit] = useState<StorageSizeUnits>(StorageSizeUnits.BYTES)
+  const [selectedUnit, setSelectedUnit] = useState<string>(StorageSizeUnits.BYTES)
```

Comment:
note that the type change here is necessary for `Select_shadcn` not to raise a type error